FIX #35766: Update buying price INT Float

gschratzer opened the pull request:
loose Decimals since V22 
Refs: Fixes #35766

Observation:
Between v21 and v22,
some updateline() call sites started adding inline casts (e.g.,
(float)$pu_ht,
 (int)$fournprice),

e.g. in compta/facture/card-rec.php. However, the model layer already normalizes monetary values before persistence, e.g.:

$sql .= ', buy_price_ht=' . ($pa_ht ? price2num($pa_ht) : 0);

// htdocs/compta/facture/class/facture-rec.class.php

Therefore, the normalization boundary should remain the model via price2num(...). Casting at call sites is redundant and—especially for money—risky ((int) truncates decimals).

Suggestion:
keep integer casts for IDs/units, and rely on a single price2num(...) in the target function for prices/rates.

Backport
If accepted, I’ll open a minimal same-change PR for v22.

gschratzer committed:
FIX #35766: Update buying price INT Float
loose Decimals

.

eldy commented on htdocs/compta/facture/class/facture-rec.class.php:
```diff
@@ -1306,7 +1306,7 @@ public function updateline($rowid, $desc, $pu_ht, $qty, $txtva, $txlocaltax1 = 0
 		$sql .= ", date_start_fill=".((int) $date_start_fill);
 		$sql .= ", date_end_fill=".((int) $date_end_fill);
-		$sql .= ", fk_product_fournisseur_price=".($fk_fournprice > 0 ? $fk_fournprice : 'null');
+		$sql .= ", fk_product_fournisseur_price=".($fk_fournprice > 0 ? price2num($fk_fournprice) : 'null');
```
fk_product_fournisseur_price is an int, the ID into table llx_product_price.

gschratzer committed:
Fix: keep fk_product_fournisseur_price as int; treat buying price as float (no truncation) [#35766]

gschratzer committed:
Updated per feedback: keep fk_product_fournisseur_price as ?int (rowid) and  as float.
Adjusted FactureRec::updateline PHPDoc to ?int  and float|''  (model still normalizes via price2num).
Tested on a recurring line: decimals preserved, FK stored as int, CI green. Minor adjustments before v22 backport

eldy commented:
Avoid using force-pushed in your pr, you must do a merge instead to not lose contribution of others.

eldy committed:
Delete .hooktest

eldy committed:
Delete dev/tools/pre-commit/README.md

eldy committed:
Delete test/phpunit/phpunit.xml
